point_cloud_test/src/lib.rs

```diff
@@ -0,0 +1,130 @@
+    pub width: f64,
+    // Height of the area to generate points for, in meters
+    pub height: f64,
+    // The number of threads used to shard octree building.
+    pub num_threads: usize,
+    // The number of points in the test point cloud.
+    pub num_points: usize,
+    // The batch size used for building the S2 point cloud.
+    pub batch_size: usize,
+    // The seed used for generating point clouds
+    pub seed: u64,
+}
+
+impl Eq for Arguments {}
+
+impl Default for Arguments {
+    fn default() -> Arguments {
+        Arguments {
+            resolution: 0.001,
+            width: 200.0,
+            height: 20.0,
+            num_threads: 10,
+            num_points: 1_000_000,
+            batch_size: 5000,
+            seed: 80_293_751_232,
+        }
+    }
+}
+
+pub fn make_octree(args: &Arguments, dir: &Path) {
+    let points_oct = SyntheticData::new(args.width, args.height, args.num_points, args.seed);
+    let bbox = points_oct.bbox();
+    let batches_oct = Batched::new(points_oct, args.batch_size);
+    let pool = scoped_pool::Pool::new(args.num_threads);
+
+    build_octree(&pool, dir, args.resolution, bbox, batches_oct, &["color"]);
+}
+
+pub fn make_s2_cells(args: &Arguments, dir: &Path) {
+    let points_s2 = SyntheticData::new(args.width, args.height, args.num_points, args.seed);
+    let mut s2_writer: S2Splitter<RawNodeWriter> =
+        S2Splitter::with_split_level(S2_LEVEL, dir, Encoding::Plain, OpenMode::Truncate);
+    Batched::new(points_s2, args.batch_size)
+        .try_for_each(|batch| s2_writer.write(&batch))
+        .expect("Writing failed");
+    let meta = s2_writer.get_meta().to_proto();
+    let mut meta_writer = BufWriter::new(File::create(dir.join("meta.pb")).unwrap());
+    meta.write_to_writer(&mut meta_writer).unwrap();
+}
+
+static INIT: Once = Once::new();
+static mut OCTREE_DIR: Option<TempDir> = None;
+static mut S2_DIR: Option<TempDir> = None;
+static mut ARGUMENTS: Option<Arguments> = None;
+
+pub fn get_s2_and_octree_path(args: &Arguments) -> (PathBuf, PathBuf, SyntheticData) {
+    let (s2_path_buf, octree_path_buf) = unsafe {
+        INIT.call_once(|| {
+            let octree_dir = TempDir::new("octree").unwrap();
+            make_octree(&args, octree_dir.path());
+            OCTREE_DIR = Some(octree_dir);
+            let s2_dir = TempDir::new("s2").unwrap();
+            make_s2_cells(&args, s2_dir.path());
+            S2_DIR = Some(s2_dir);
+            ARGUMENTS = Some(args.clone());
+        });
+        // If somebody called this twice with different arguments, caching doesn't make sense
+        assert_eq!(ARGUMENTS, Some(args.clone()));
+        let s2_path_buf = S2_DIR.as_ref().unwrap().path().to_owned();
+        let octree_path_buf = OCTREE_DIR.as_ref().unwrap().path().to_owned();
+        (s2_path_buf, octree_path_buf)
+    };
+    let data = SyntheticData::new(10.0, 10.0, args.num_points, args.seed);
```

Comment:
nit: would it be helpful to move the numbers into costants?

Reply:
Actually, these should be args.width and args.height. Fixed.